azure-pipelines.yml

```diff
@@ -24,6 +24,52 @@ variables:
     openblas_version: 0.3.9
 
 jobs:
+- job: pre_release_deps_source_dist
+  condition: and(succeeded(), ne(variables['Build.SourceBranch'], 'refs/heads/master'))  # skip for PR merges
+  pool:
+    vmImage: 'ubuntu-18.04'
+  steps:
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: '3.7'
+      addToPath: true
+      architecture: 'x64'
+  - script: >-
+      python -m pip install --upgrade
+      pip==20.2.4
+      "setuptools<50"
+      wheel
+    displayName: 'Install deps'
+    failOnStderr: true
+  - script: >-
+      python -m pip install --pre --upgrade --timeout=60
+      cython
+      mpmath
+      pybind11
+      pytest
+      pytest-xdist
+    displayName: 'Install pre-release deps'
+    failOnStderr: true
+  - script: |
+      python -m pip install --pre --upgrade --timeout=60 -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy
+    displayName: 'Install pre-release NumPy wheel'
+    failOnStderr: true
+  - script: |
+        python tools/suppress_output.py python setup.py sdist
+        cd dist
+        python -m pip install $PWD/scipy* -v
```

Comment:
Maybe the build isolation is causing issues for the build? I think the Travis CI comments suggest build isolation is desirable for this CI entry.

Reply:
I'll try to debug on my fork, it may be the sdist setup proper